src/notifications/useFetchNotifications.ts

```diff
@@ -0,0 +1,101 @@
+import { useCallback, useState } from 'react';
+
+import { useDispatch, useSelector } from 'react-redux';
+
+import { useHeadlessService } from './useHeadlessService';
+import useMarkNotificationAsSeen from './useMarkNotificationAsSeen';
+
+import {
+  selectLoadedNotificationsPages,
+  selectNotificationsIsFetching,
+  setNotificationsLoading,
+  setNotificationsPageAndFinishLoading,
+} from '@/redux/slices/notifications';
+import { areArraysEqual } from '@/utils/array';
+
+const NOTIFICATIONS_PAGE_SIZE = 10;
+
+const useFetchNotifications = () => {
```

Comment:
As discussed, something causes an infinite loop of calling `fetch`

Reply:
Turns out it was `useHeadlessService` that was infinitely initializing. It was fixed.